File: app/src/types/recruitment.ts
export type Role = "hr" | "candidate";

export type ResumeProject = {
  name: string;
  techStack: string[];
  impact: string;
  link?: string;
};

export type ResumeCandidate = {
  id: string;
  candidateName: string;
  jobRole: string;
  atsScore: number;
  extractedSkills: string[];
  certifications: string[];
  achievements: string[];
  experience: string;
  suitability: number;
  status: "pending" | "selected" | "interview_completed" | "rejected";
  authorizationStatus?: "pending" | "authorized" | "portal_accessed";
  email: string;
  phone?: string;
  location?: string;
  education?: string[];
  languages?: string[];
  professionalSummary?: string;
  projects?: ResumeProject[];
  zoomLink?: string;
  linkedinUrl?: string;
  githubUrl?: string;
  portfolioUrl?: string;
  currentCompany?: string;
  profilePhotoUrl?: string;
  resumeFileUrl?: string;
};

export type JobPosting = {
  id: string;
  title: string;
  requiredAts: number;
  jdFileName: string;
  jdFileUrl?: string;
  applySlug?: string;
  applyLink?: string;
  isActive?: boolean;
};

export type JobApplicationPayload = {
  fullName: string;
  email: string;
  phone?: string;
  location?: string;
  totalExperience?: string;
  currentCompany?: string;
  linkedinUrl?: string;
  portfolioUrl?: string;
  education?: string;
  professionalSummary?: string;
  resume: File;
  profilePhoto?: File | null;
};

export type PublicJobPosting = {
  id: string;
  title: string;
  jdFileName: string;
  jdFileUrl?: string;
  applySlug: string;
};

File: backend/app/schemas/recruitment.py
from typing import Literal

from pydantic import BaseModel, ConfigDict, EmailStr


CandidateStatus = Literal["pending", "selected", "interview_completed", "rejected"]
AuthorizationStatus = Literal["pending", "authorized", "portal_accessed"]


class ResumeProject(BaseModel):
    model_config = ConfigDict(extra="forbid")

    name: str
    techStack: list[str]
    impact: str
    link: str | None = None


class Candidate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    candidateName: str
    jobRole: str
    atsScore: int
    extractedSkills: list[str]
    certifications: list[str]
    achievements: list[str]
    experience: str
    suitability: int
    status: CandidateStatus
    authorizationStatus: AuthorizationStatus = "pending"
    email: EmailStr
    phone: str | None = None
    location: str | None = None
    education: list[str] | None = None
    languages: list[str] | None = None
    professionalSummary: str | None = None
    projects: list[ResumeProject] | None = None
    zoomLink: str | None = None
    linkedinUrl: str | None = None
    githubUrl: str | None = None
    portfolioUrl: str | None = None
    currentCompany: str | None = None
    profilePhotoUrl: str | None = None
    resumeFileUrl: str | None = None


class CandidateListResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    items: list[Candidate]


class CandidateBulkUpsertRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    items: list[Candidate]


class CandidateAuthorizationRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    candidateId: str


class CandidateAuthorizationResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    success: bool
    message: str
    candidateId: str
    email: str | None = None
    password: str | None = None


class JobPosting(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    title: str
    requiredAts: int
    jdFileName: str
    jdFileUrl: str | None = None
    applySlug: str = ""
    applyLink: str = ""
    isActive: bool = True


class JobPostingListResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    items: list[JobPosting]


class JobPostingBulkUpsertRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    items: list[JobPosting]


class JobPostingPublicResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    title: str
    jdFileName: str
    jdFileUrl: str | None = None
    applySlug: str
